src/test/java/com/amazonaws/services/dynamodbv2/AmazonDynamoDBLockClientTest.java

```diff
@@ -235,6 +235,37 @@ public void acquireLock_withAcquireOnlyIfLockAlreadyExists_releasedLockGetsCreat
         assertEquals("asdf", lockItem.getPartitionKey());
     }
 
+    @Test
+    public void acquireLock_withReentrant_doesNotFailIfHoldingLock() throws InterruptedException {
```

Comment:
Please make sure we have a test where the second call to acquire lock fails. If it's not there already, please add it